### PR TITLE
added dates calculator functions, templates and styles

### PR DESCRIPTION
**Pull request by liaarbel**
I created a form for two dates. One is for the first date we start from and the second is for the date we end with.
After clicking on the button "calculate" we get the difference in days between the two dates.
We don't have to enter data to the first date input, what will happen is that the first date will become the current date.

### CODE REVIEW

File: app/templates/home.html [hunk 1]
```diff
@@ -1,21 +1,56 @@
 {% extends "partials/index/index_base.html" %}
 
 {% block content %}
+<script>
+        function hiddenDifference() {
+            if (document.getElementById("endDate").value == '') {
+                alert("Expected end date");
```
Comment (yammesicka) [inline app/templates/home.html]: Please don't use `alert`, use models or sweetalert instead

File: app/templates/home.html [hunk 2]
```diff
@@ -1,21 +1,56 @@
...
+                return;
+            }
+            var date1 = document.getElementById("startDate").value;
```
Comment (yammesicka) [inline app/templates/home.html]: Use let/const

File: app/templates/home.html [hunk 3]
```diff
@@ -1,21 +1,56 @@
...
+            var date2 = new Date(document.getElementById("endDate").value);
+            if (date1 != '') {
+                date1 = new Date(date1);
+                const diffTime = Math.abs(date2 - date1);
+                const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+                document.getElementById("demo").innerHTML = "The difference: " + (diffDays) + " days";
```
Comment (yammesicka) [inline app/templates/home.html]: These 3 lines can be outside the `if`

File: app/templates/home.html [hunk 4]
```diff
@@ -1,21 +1,56 @@
 {% extends "partials/index/index_base.html" %}
 
 {% block content %}
+<script>
```
Comment (yammesicka) [inline app/templates/home.html]: Please put the JS in a different file

**Review by yammesicka — CHANGES_REQUESTED** [review]
That's a great start :)

File: app/templates/home.html (hunk 1 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html (hunk 2 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html (hunk 3 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html (hunk 4 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html [hunk 5]
```diff
@@ -1,21 +1,34 @@
 {% extends "partials/index/index_base.html" %}
 
 {% block content %}
+<script src="https://unpkg.com/sweetalert/dist/sweetalert.min.js"></script>
+<script src="{{ url_for('static', path='/js/dates_calculator.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/home.html]: Can you append this to the `<head>`?

File: app/templates/home.html [hunk 6]
```diff
@@ -1,21 +1,34 @@
...
+<div class="form-style">
+    <h3>Time calculator:</h3>
+    <form>
+        <label for="startDate">Start date:</label>
+        <input id="startDate" name="date1" type="date">
+        <br>
+        <label for="endDate">End date:<em>&#x2a;</em></label>
+        <input id="endDate" name="date2" type="date">
+    </form>
+    <button type="button" onclick="hiddenDifference()">Calculate</button>
```
Comment (yammesicka) [inline app/templates/home.html]: Please don't use `onclick` - Use `addEventListener` instead. (We don't want to mix the logic and the view)

File: app/static/js/dates_calculator.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    const date2 = new Date(document.getElementById("endDate").value);
+    if (date1 != '') {
+        date1 = new Date(date1);
+    }
+    else {
+        date1 = Date.now()
+    }
+    const diffTime = Math.abs(date2 - date1);
+    const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+    document.getElementById("demo").innerHTML = "The difference: " + (diffDays) + " days";
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js]: Prefer `innerText`

File: app/static/js/dates_calculator.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    let date1 = document.getElementById("startDate").value;
+    const date2 = new Date(document.getElementById("endDate").value);
+    if (date1 != '') {
+        date1 = new Date(date1);
+    }
+    else {
+        date1 = Date.now()
+    }
+    const diffTime = Math.abs(date2 - date1);
+    const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js]: Save as a constant with a proper name

File: app/templates/home.html [hunk 9]
```diff
@@ -1,21 +1,34 @@
...
+    <h3 id="demo"></h3>
```
Comment (yammesicka) [inline app/templates/home.html]: What is the purpose of the `h3`? Should it be a title?

If not - prefer using `p`/`span`/`div` tags + CSS

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/home.html (hunk 5 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html (hunk 9 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/static/js/dates_calculator.js (hunk 8 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js]: fixed

File: app/templates/home.html [hunk 10]
```diff
@@ -1,21 +1,34 @@
 {% extends "partials/index/index_base.html" %}
 
 {% block content %}
+<script src="https://unpkg.com/sweetalert/dist/sweetalert.min.js"></script>
+<script src="{{ url_for('static', path='/js/dates_calculator.js') }}"></script>
 
+<div class="container mt-4">
+</div>
```
Comment (yammesicka) [inline app/templates/home.html]: What is the purpose of this div?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/static/js/dates_calculator.js (hunk 7 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js]: fixed

File: app/templates/home.html (hunk 6 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/home.html (hunk 10 above)
Comment (liaarbel) [reply app/templates/home.html]: fixed

File: app/templates/calendar_monthly_view.html [hunk 11]
```diff
@@ -5,6 +5,17 @@
       <div class="title">{{ day.display() }}</div>
       <div class="sec-title">Location 0<sup>o</sup>c 00:00</div>
     </div>
+    <div style="align: center">
```
Comment (yammesicka) [inline app/templates/calendar_monthly_view.html]: Don't use inline styles, move them to CSS file

File: app/templates/home.html (hunk 5 above)
Comment (yammesicka) [reply app/templates/home.html]: Was it?

File: app/templates/home.html (hunk 10 above)
Comment (yammesicka) [reply app/templates/home.html]: ?

File: app/templates/partials/calendar/calendar_base.html [hunk 12]
```diff
@@ -22,4 +22,6 @@
 <!-- Scripts -->
 <script src="{{ url_for('static', path='/js/grid_scripts.js') }}"></script>
 <script src="{{ url_for('static', path='/js/grid_navigation.js') }}"></script>
+<script src="https://unpkg.com/sweetalert/dist/sweetalert.min.js"></script>
+<script src="{{ url_for('static', path='/js/dates_calculator.js') }}"></script>
```
Comment (yammesicka) [inline app/templates/partials/calendar/calendar_base.html:23]: You should load it only on your specific page. Use [blocks](https://jinja.palletsprojects.com/en/2.11.x/templates/#template-inheritance) to inject it elegantly

File: app/static/js/dates_calculator.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    const date2 = new Date(document.getElementById("endDate").value);
+    if (date1 != '') {
+        date1 = new Date(date1);
+    }
+    else {
+        date1 = Date.now()
+    }
+    const diffDates = Math.abs(date2 - date1);
+    const diffInDays = Math.ceil(diffDates / (1000 * 60 * 60 * 24));
+    document.getElementById("demo").innerText = "The difference: " + (diffInDays) + " days";
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js:20]: You can use JS ``` `Template ${strings}` ```  if you'd like to

File: app/static/js/dates_calculator.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+        swal("Expected end date");
+        return;
+    }
+    let date1 = document.getElementById("startDate").value;
+    const date2 = new Date(document.getElementById("endDate").value);
+    if (date1 != '') {
+        date1 = new Date(date1);
+    }
+    else {
+        date1 = Date.now()
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js]: ;

File: app/static/js/dates_calculator.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
+window.onload = function () {
+  document.getElementById("CalcBtn").addEventListener("click", hiddenDifference);
+}
+
+function hiddenDifference() {
+    if (document.getElementById("endDate").value == '') {
+        swal("Expected end date");
+        return;
+    }
+    let date1 = document.getElementById("startDate").value;
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js:10]: Prefer `const`

File: app/static/js/dates_calculator.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
+window.onload = function () {
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js]: Prefer addEventListener on DOMContentLoaded to prevent the possibility of being overriden

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/templates/home.html (hunk 10 above)
Comment (liaarbel) [reply app/templates/home.html]: It doesn't exist in the new commit

File: app/static/js/dates_calculator.js (hunk 15 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js:10]: but `date1` changing

File: app/static/js/dates_calculator.js (hunk 16 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js]: I really don't understand how

File: app/static/js/dates_calculator.js (hunk 14 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js]: fixed

File: app/static/js/dates_calculator.js (hunk 13 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js:20]: I took this [code](https://stackoverflow.com/questions/42341761/javascript-eventlistener-not-working-in-external-js-file) and I really don't know or want to change something that work pretty code :(

File: app/templates/calendar_monthly_view.html (hunk 11 above)
Comment (liaarbel) [reply app/templates/calendar_monthly_view.html]: fixed

File: app/static/js/dates_calculator.js (hunk 13 above)
Comment (yammesicka) [reply app/static/js/dates_calculator.js:20]: No problem, but we did talk about taking code from others. Please include full credit in your code.

File: app/templates/calendar_monthly_view.html (hunk 11 above)
Comment (yammesicka) [reply app/templates/calendar_monthly_view.html]: It is still here :)

File: app/templates/home.html (hunk 10 above)
Comment (yammesicka) [reply app/templates/home.html]: I can't see it here. Is there a new PR or something?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/static/js/dates_calculator.js [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+    let date1 = document.getElementById("startDate").value;
+    const date2 = new Date(document.getElementById("endDate").value);
+    if (date1 != '') {
+        date1 = new Date(date1);
+    }
+    else {
+        date1 = Date.now();
+    }
+    const diffDates = Math.abs(date2 - date1);
+    const diffInDays = Math.ceil(diffDates / (1000 * 60 * 60 * 24));
```
Comment (ivarshav) [inline app/static/js/dates_calculator.js:19]: maybe set the values to a const?

File: app/templates/home.html (hunk 10 above)
Comment (liaarbel) [reply app/templates/home.html]: Every time I pull from the develop this div exists

File: app/static/js/dates_calculator.js (hunk 17 above)
Comment (liaarbel) [reply app/static/js/dates_calculator.js:19]: It is const

Comment (ivarshav) [reply app/static/js/dates_calculator.js:19]: Sorry, I meant the *numbers* - `1000 * 60 * 60 * 24`, extract it to another const.

Comment (liaarbel) [reply app/static/js/dates_calculator.js:19]: Is that what matters? It's just a formula
Here you can see the logical in the calculation

File: app/static/js/dates_calculator.js [hunk 18]
```diff
@@ -0,0 +1,21 @@
+window.addEventListener('DOMContentLoaded', (event) => {
+    document.getElementById("CalcBtn").addEventListener("click", hiddenDifference);
+});
```
Comment (yammesicka) [inline app/static/js/dates_calculator.js:3]: Great job!

**Review by yammesicka — APPROVED** [review]

**Pull request merged**